modules/videoio/src/cap_gstreamer.cpp

```diff
@@ -1414,7 +1414,18 @@ bool CvVideoWriter_GStreamer::open( const char * filename, int fourcc,
         g_object_set(G_OBJECT(file), "location", filename, NULL);
     }
 
-    if (is_color)
+    if (fourcc == CV_FOURCC('M','J','P','G') && frameSize.height == 1)
+    {
+        input_pix_fmt = GST_VIDEO_FORMAT_ENCODED;
+#if GST_VERSION_MAJOR > 0
+        caps = gst_caps_new_simple("image/jpeg",
+                                   "framerate", GST_TYPE_FRACTION, int(fps), 1,
+                                   NULL);
+        caps = gst_caps_fixate(caps);
+
+#endif
```

Comment:
I believe "#else" branch should be handled somehow (error reporting for example).